client/core/core.go

```diff
@@ -208,6 +216,82 @@ func (c *Core) handleReconnect(uri string) {
 	log.Infof("DEX at %s has reconnected", uri)
 }
 
+// handleOrderBookMsg is called when an orderbook request is received.
+func (c *Core) handleOrderBookMsg(dc *dexConnection, msg *msgjson.Message) error {
+	resp, err := msg.Response()
+	if err != nil {
+		return err
+	}
+
+	var snapshot msgjson.OrderBook
+	err = json.Unmarshal(resp.Result, &snapshot)
+	if err != nil {
+		return fmt.Errorf("order book unmarshal error: %v", err)
+	}
+
+	ob := order.NewOrderBook()
+	err = ob.Sync(&snapshot)
+	if err != nil {
+		return err
+	}
+
+	dc.booksMtx.Lock()
+	dc.books[snapshot.MarketID] = ob
+	dc.booksMtx.Unlock()
```

Comment:
Value in checking `snapshot.MarketID != ""` immediately after Unmarshal? An empty string is a valid map key value but it's almost certainly a problem.